fiftyone/utils/labelstudio.py

```diff
@@ -0,0 +1,835 @@
+                else labels[0]
+            )
+            sample_id = task_id2sample_id[t["id"]]
+            results[sample_id] = label_ids
+
+        return results
+
+    def _export_to_labelstudio(self, labels, label_type):
+        if _LABEL_TYPES[label_type]["multiple"] is None:
+            return export_label_to_labelstudio(labels)
+        return [export_label_to_labelstudio(l) for l in labels]
+
+    def upload_tasks(self, samples, backend):
+        """Create a Label Studio project and upload samples"""
+        config = backend.config
+        project = self._init_project(config, samples)
+        tasks, predictions, id_map = self._prepare_tasks(
+            samples, config.label_schema
+        )
+        uploaded_tasks = self._import_tasks(project, tasks, predictions)
+        return LabelStudioAnnotationResults(
+            samples,
+            config,
+            id_map=id_map,
+            project_id=project.id,
+            uploaded_tasks=uploaded_tasks,
+            backend=backend,
+        )
+
+    def download_annotations(self, results):
+        """Import annotations from Label Studio.
+
+        Args:
+            results: LabelStudioAnnotationResults object
+
+        Returns:
+            # Scalar fields
+            results[label_type][sample_id] = scalar
+
+            # Label fields
+            results[label_type][sample_id][label_id] = label
+        """
+        project = self.get_project(results.project_id)
+        labelled_tasks = self._get_matched_labelled_tasks(
+            project, list(results.uploaded_tasks.keys())
+        )
+        annotations = {}
+        for label_field, label_info in results.config.label_schema.items():
+            return_type = foua._RETURN_TYPES_MAP[label_info["type"]]
+            labels = self._import_annotations(
+                labelled_tasks, results.uploaded_tasks, return_type
+            )
+            annotations.update({label_field: {return_type: labels}})
+        return annotations
+
+    def upload_predictions(self, project, tasks, sample_labels, label_type):
+        """Upload existing labels to Label Studio."""
+        for task, labels in zip(tasks, sample_labels):
+            predictions = self._export_to_labelstudio(labels, label_type)
+            project.create_prediction(task, predictions)
+
+    def delete_tasks(self, task_ids):
+        """Delete tasks from Label Studio.
+
+        Args:
+            task_ids: list of task ids
+        """
+        for t_id in task_ids:
+            self.make_request(
+                "DELETE",
+                f"/api/tasks/{t_id}",
+            )
+
+    def delete_project(self, project_id):
+        """Delete project from Label Studio.
+
+        Args:
+            project_id: project id
+        """
+        self.make_request(
+            "DELETE",
+            f"/api/projects/{project_id}",
+        )
+
+
+class LabelStudioAnnotationResults(foua.AnnotationResults):
```

Comment:
Note, that this class requires the [`_from_dict()`](https://github.com/voxel51/fiftyone/blob/8cab046e7fe98ad04e37bab70183e03e128904bf/fiftyone/utils/annotations.py#L2094) method to allow the run results to be loaded properly from the database. You can see an example of it [here](https://github.com/voxel51/fiftyone/blob/8cab046e7fe98ad04e37bab70183e03e128904bf/fiftyone/utils/labelbox.py#L1378)